src/prefect/tasks/gcp/secretmanager.py

```diff
@@ -0,0 +1,87 @@
+    (https://googleapis.dev/python/google-api-core/latest/auth.html#explicit-credentials),
+    or otherwise it will use [default Google client logic]
+    (https://googleapis.dev/python/google-api-core/latest/auth.html).
+
+    Args:
+        - project_id (Union[str, int], optional): the name of the project where the Secret is saved
+        - secret_id (str, optional): the name of the secret to retrieve
+        - version_id (Union[str, int], optional): the version number of the secret to use;
+            defaults to 'latest'
+        - credentials (dict, optional): dictionary containing GCP credentials
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        project_id: Union[str, int] = None,
+        secret_id: str = None,
+        version_id: Union[str, int] = "latest",
+        credentials: dict = None,
+        **kwargs,
+    ):
+        self.project_id = project_id
+        self.secret_id = secret_id
+        self.version_id = version_id
+        self.credentials = credentials
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("project_id", "secret_id", "version_id", "credentials")
+    def run(
+        self,
+        project_id: Union[str, int] = None,
+        secret_id: str = None,
+        version_id: Union[str, int] = "latest",
+        credentials: dict = None,
+    ) -> dict:
+        """
+        Task run method.
+
+        Args:
+            - project_id (Union[str, int], optional): the name of the project where the Secret is saved
+            - secret_id (str, optional): the name of the secret to retrieve
+            - version_id (Union[str, int], optional): the version number of the secret to use;
+                defaults to "latest"
+            - credentials (dict, optional): your GCP credentials passed from an upstream Secret task.
+                If not provided here [default Google client logic]
+                (https://googleapis.dev/python/google-api-core/latest/auth.html) will be used.
+
+        Returns:
+            - dict: the contents of this secret, as a dictionary
+        """
+        if project_id is None:
+            raise ValueError("A GCP project ID must be provided.")
+        if secret_id is None:
+            raise ValueError("A GCP Secret Manager secret ID must be provided.")
+
+        # Create the Secret Manager client.
+        client = secretmanager.SecretManagerServiceClient(credentials=credentials)
+
+        # Build the resource name of the secret version.
+        name = f"projects/{project_id}/secrets/{secret_id}/versions/{version_id}"
+
+        # Access the secret version.
+        response = client.access_secret_version(name=name)
+
+        # Return the decoded payload.
+        secret_string = response.payload.data.decode("UTF-8")
+
+        secret_dict = json.loads(secret_string)
```

Comment:
I don't think we want to load from json here, gcp secrets can have other data types, we should just return the string for now